core/src/main/scala/org/apache/spark/api/python/PythonWorkerFactory.scala

```diff
@@ -184,10 +185,20 @@ private[spark] class PythonWorkerFactory(
       redirectStreamsToStderr(workerProcess.getInputStream, workerProcess.getErrorStream)
 
       // Wait for it to connect to our socket, and validate the auth secret.
-      serverSocketChannel.socket().setSoTimeout(10000)
```

Comment:
There was supposed to be 10 second timeout. But this call does not seem to affect `serverSocketChannel.accept()`.  
This set might only take effect if we did `serverSocketChannel.socket().accept()`, but hat returns a socket, not a channel.